### DISCUSSION TITLE
How can I replace a device on node with metadatadevice configured?

### DISCUSSION THREAD

**Discussion by CrossainQi** [rook/rook#12130] [Q&A]
Hi all, I am testing replacing device on node with metadatadevice configured but got some problems. Here are my steps:
replacing data device(osd):
- kubectl -n rook-ceph scale deployment rook-ceph-operator --replicas=0
- osd-purge.yaml
- replace the device
- edit the cluster.yaml and apply ( with new lv as metadatadevice )
- kubectl -n rook-ceph scale deployment rook-ceph-operator --replicas=1

replacing db device:
- kubectl -n rook-ceph scale deployment rook-ceph-operator --replicas=0
- change the osd's replicas that using this db device to 0
- replace the device 
- edit the cluster.yaml to indicate new db device
- kubectl -n rook-ceph scale deployment rook-ceph-operator --replicas=1

Is there any problems in my steps?

**Comment by CrossainQi** [comment]
The problem I faced during replacing db device can be seen [here](https://github.com/rook/rook/issues/12122#issue-1678100265).

**Reply by travisn** [reply]
@CrossainQi What is different about this discussion from #12122? Or is it the same issue?

**Reply by CrossainQi** [reply]
yes, they are the same issues.

**Comment by CrossainQi** [comment]
When I tried to replace an data device(osd), the newly created osd-1 crashed during creation, the osd log was as follows:
```log
k logs -f -n rook-ceph rook-ceph-osd-1-978cb4cbc-ps2bq osd
debug 2023-04-24T03:44:40.410+0000 7fafd016c700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
debug 2023-04-24T03:44:40.410+0000 7fafcf96b700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
debug 2023-04-24T03:44:40.410+0000 7fafd096d700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
failed to fetch mon config (--no-mon-config to skip)
```
